cnap/userv/streaming.py

```diff
@@ -310,12 +311,15 @@ def execute(self):
             self.infer_queue_connector.publish_frame(
                 self.infer_engine_info.queue_topic, frame)
 
+            end_time = time.time()
+            processing_time = end_time - start_time
+
             # 6. Prepare next loop
             if isinstance(self.provider, FileSource):
                 if self.provider.target_fps != -1:
-                    time.sleep(float(1/self.provider.target_fps))
+                    time.sleep(max(float(1/self.provider.target_fps) - processing_time, 0))
                 else:
-                    time.sleep(float(1/self.provider.raw_fps))
+                    time.sleep(max(float(1/self.provider.raw_fps) - processing_time, 0))
```

Comment:
It makes sense. But what we should think more:
- what if `float(1/self.provider.raw_fps) - processing_time` < 0, which means the streaming service can not meet the requirments, we should scale it based on VPA, becasue increase replicas is not suitale in this case.
- Maybe try to accelerate using qat, amx, dsa to minimal the process time.

Reply:
Thanks comment, it makes sense.